### PR TITLE
Workflow Updates

### PR DESCRIPTION
**Pull request by dmitri-mcguckin**
# Features
* Add a workflow for building and publishing the maven package to the Github Maven repository
* Add a workflow for running unit tests

### CODE REVIEW

**Comment by buggtb** [comment]
Does this jar end up anywhere other than the github maven repo?

**Comment by buggtb** [comment]
Also, just from a build / JDK support perspective, whats the point of the Nashorn exceptions? If you need some specific exception types that don't exist in the default JDK why not just build them? Cause we end up in a situation where they either don't exist, or we're pulling in a Javascript engine purely to provide some form of Java based exception handling. I'm not sure what the reasoning is there.

**Comment by dmitri-mcguckin** [comment]
@buggtb the nashhorn stuff are not exceptions, they're decorators, and they've been used for a bit to mark deprecated code.

Also, for now, it only goes to the github repository, I spent some time figuring out the whole deploy scheme for deploying to the Maven Central Repo it involves setting up am OSSH account and all. So for the time being I'm just playing around with setting it up here.


Github's Maven repos seem to work fine, the only minor annoyance is that you have to specify the repo url for Maven to resolve it.

**Comment by buggtb** [comment]
The problem with github and maven, is to use the github maven repo, you have to specify a public token for your github account to build it. Which is why I used Gitlab, because you don't. As this lives both in this repo and I've been asked by Matt to ensure I get this sync'd upstream ASAP, we shouldn't be merging in a PR that requires a repo authentication because then upstream users will have a dependency on a working github account in their CI pipelines, and ours.

**Comment by dmitri-mcguckin** [comment]
I'm not sure what you mean by the github-user building it. The token used is auto-generated by github at the time the runner is spawned, and ostensibly is unique for every new runner (every time a workflow runs).

There's no manual generation of any token, the workflow could be transplanted to a wholly new repo here on github and still work.

File: pom.xml [hunk 1]
```diff
@@ -20,30 +20,82 @@
...
+        <!-- AShot Repo -->
         <repository>
             <id>gitlab-maven-ashot</id>
             <url>https://gitlab.com/api/v4/projects/28250236/packages/maven</url>
         </repository>
-    </repositories>
 
-    <distributionManagement>
+        <!-- Clojars Repo -->
         <repository>
```
Comment (buggtb) [inline pom.xml:30]: I think this and the bit above still needs reverting cause the CI job is failing.

**Review by buggtb — CHANGES_REQUESTED** [review]

**Review by buggtb — APPROVED** [review]

**Pull request merged**